pkg/node/node.go

```diff
@@ -363,3 +434,112 @@ func jsonPatchEscape(value string) string {
 	value = strings.Replace(value, "~", "~0", -1)
 	return strings.Replace(value, "/", "~1", -1)
 }
+
+func addTaint(node *corev1.Node, nth Node, taintKey string, taintValue string, effect corev1.TaintEffect) error {
```

Comment:
This, `addTaintToSpec` and `cleanTaint` are lifted from `cluster-autoscaler` with minor modifications.